src/main/java/com/derongan/minecraft/mineinabyss/geary/AbyssLocationSystem.kt

```diff
@@ -0,0 +1,64 @@
+package com.derongan.minecraft.mineinabyss.geary
+
+import com.derongan.minecraft.deeperworld.world.section.Section
+import com.derongan.minecraft.mineinabyss.AbyssContext
+import com.derongan.minecraft.mineinabyss.world.Layer
+import com.mineinabyss.geary.ecs.Family
+import com.mineinabyss.geary.ecs.component.components.control.Activated
+import com.mineinabyss.geary.ecs.entity.GearyEntity
+import com.mineinabyss.geary.ecs.system.IteratingSystem
+import org.bukkit.ChatColor
+import org.bukkit.Location
+
+class AbyssLocationSystem(private val context: AbyssContext) : IteratingSystem(Family.builder().setAll(setOf(Activated::class.java, DepthMeter::class.java)).build()) {
```

Comment:
should this be in deeper world?

Reply:
I think DeeperWorld should only hold a global coordinate system, the abyssal depth seems pretty specific to us. Though once we have a system like that in DeeperWorld, we could probably use it here for layer depth calculations.

Reply:
Offz is correct